scheduler/plugin/wasm.go

```diff
@@ -0,0 +1,73 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package wasm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+// getURL parses the URL manually, so that it can resolve relative file paths,
+// such as "file://../../path/to/plugin.wasm"
+func getURL(ctx context.Context, url string) ([]byte, error) {
+	if url == "" {
+		return nil, errors.New("missing url")
```

Comment:
This error is going to appear to users. So, can we make it a bit more fancy like "GuestURL is required in the plugin configuration"?

Reply:
I moved the missing one higher up and backfilled a test, so that they are like the others (prefixed `wasm: some error`)